web/src/main/java/org/mskcc/cbio/oncokb/api/pub/v1/TrialsApiController.java

```diff
@@ -0,0 +1,360 @@
+            }
+
+            if (result.isEmpty()){
+                status = HttpStatus.NOT_FOUND;
+                return new ResponseEntity<>(result, status);
+            }
+        }
+        return new ResponseEntity<>(result, status);
+    }
+
+    private Tumor getTumor(JSONObject jsonObject, String oncoTreeCode){
+        Tumor tumor = new Tumor();
+        if (jsonObject.containsKey(oncoTreeCode)){
+            JSONObject tumorObj = (JSONObject)jsonObject.get(oncoTreeCode);
+            tumor.setNciCode((String)tumorObj.get("nciCode"));
+            tumor.setNciMainType((String)tumorObj.get("nciMainType"));
+            tumor.setTumorName((String)tumorObj.get("tumorName"));
+    
+            JSONArray trialsArr = (JSONArray)tumorObj.get("trials");
+            Trial[] trials = getTrials(trialsArr);
+            tumor.setTrials(trials);
+        }
+
+        return tumor;
+    }
+
+    private Trial[] getTrials(JSONArray trialsArr){
+        Trial[] trials = new Trial[trialsArr.size()];
+        for (int i = 0; i < trialsArr.size(); i++){
+            JSONObject trial = (JSONObject)trialsArr.get(i);
+            trials[i] = new Trial();
+            trials[i].setBriefTitle((String)trial.get("briefTitle"));
+            trials[i].setCurrentTrialStatus((String)trial.get("currentTrialStatus"));
+            trials[i].setNctId((String)trial.get("nctId"));
+            if (trial.containsKey("isUSTrial"))
+                trials[i].setIsUSTrial((Boolean)trial.get("isUSTrial"));
+            if (trial.containsKey("principalInvestigator"))
+                trials[i].setPrincipalInvestigator((String)trial.get("principalInvestigator"));
+
+            JSONArray armsArr = (JSONArray)trial.get("arms");
+            Arms[] arms = getArms(armsArr);           
+            trials[i].setArms(arms);
+        }
+        return trials;
+    }
+
+    private Arms[] getArms(JSONArray armsArr){
+        Arms[] arms = new Arms[armsArr.size()];
+        for (int j = 0; j < armsArr.size(); j++){
+            JSONObject arm = (JSONObject)armsArr.get(j);
+            arms[j] = new Arms();
+            arms[j].setArmDescription((String)arm.get("armDescription"));
+
+            JSONArray drugsArr = (JSONArray)arm.get("drugs");
+            Drug[] drugs = getDrugs(drugsArr);
+            arms[j].setDrugs(drugs);
+        }
+        return arms;
+    }
+
+    private Drug[] getDrugs(JSONArray drugsArr){
+        Drug[] drugs = new Drug[drugsArr.size()];
+        for (int k = 0; k < drugsArr.size(); k++){
+            JSONObject drug = (JSONObject)drugsArr.get(k);
+            drugs[k] = new Drug();
+            drugs[k].setDrugName((String)drug.get("drugName"));
+            drugs[k].setNcitCode((String)drug.get("ncitCode"));
+        }
+        return drugs;
+    }
+
+    private Trial[] getTrialByTumorTreatment(Trial[] trials, String treatment){
+        List<Trial> res = new ArrayList<>();
+        String[] tempDrugName = treatment.split(",|\\+");

```

Comment:
why are we matching back slash?

Reply:
"\\"is java escape character, '+' needs escape character to be identify.